system/Images/Handlers/GDHandler.php

```diff
@@ -451,7 +451,7 @@ protected function textOverlay(string $text, array $options = [], bool $isShadow
          * Get the rest of the string and split it into 2-length
          * hex values:
          */
-        $opacity = ($options['opacity'] * 127);
+        $opacity = (int) ($options['opacity'] * 127);
```

Comment:
Casting by `(int)` removes the decimals part. https://3v4l.org/8lXKD
Is that the intended behavior or should we use `round()`?

Reply:
Looking at my comment in https://github.com/codeigniter4/CodeIgniter4/pull/5370#issuecomment-976165819, it seems this is to match the current behavior.

Reply:
Frankly, I don't think there is much difference either way.
And yes, this is to match the current behavior.